components/server/src/server.rs

```diff
@@ -1204,6 +1224,77 @@ impl<ER: RaftEngine> TiKvServer<ER> {
+                    quota_limiter.set_background_cpu_time_limit(
+                        1000 * cmp::max(
+                            BACKGROUND_REQUEST_CORE_LOWER_BOUND,
+                            (SysQuota::cpu_cores_quota() * BACKGROUND_REQUEST_CORE_MAX_RATIO)
+                                as usize,
+                        ),
+                    );
+
+                    info!("cpu_time_limiter tuned for backend request";
+                            "cpu_util" => ? 0.5f64,
+                            "new quota" => ?quota_limiter.background_cputime_limiter() / 1_000.0);
+                } else {
+                    let cpu_usage = match proc_stats.cpu_usage() {
+                        Ok(r) => r,
+                        Err(_e) => 0.0,
+                    };
+
+                    // Try tuning quota when cpu_usage is correctly collected.
+                    // rule based tuning:
+                    //      1) if instance is busy, shrink cpu quota for analyze by 1 core until lower bound is hit;
+                    //      2) if instance cpu usage is at healthy state, no op;
+                    //      3) if instance is idle, increase cpu quota by 1 core until upper bound is hit.
+                    if cpu_usage > 0.0f64 {
+                        let cpu_util = (cpu_usage / 100.0) / SysQuota::cpu_cores_quota();
+
+                        let old_quota = quota_limiter.background_cputime_limiter();
+
+                        if cpu_util >= SYSTEM_BUSY_THRESHOLD {
+                            quota_limiter.set_background_cpu_time_limit(
+                                1_000
+                                    * cmp::max(
+                                        (quota_limiter.background_cputime_limiter() / 1_000_000.0)
+                                            as usize
+                                            - 1,
+                                        BACKGROUND_REQUEST_CORE_LOWER_BOUND,
+                                    ),
+                            );
+                        } else if cpu_util < SYSTEM_HEALTHY_THRESHOLD {
+                            quota_limiter.set_background_cpu_time_limit(
+                                1_000
+                                    * cmp::min(
+                                        (quota_limiter.background_cputime_limiter() / 1_000_000.0)
+                                            as usize
+                                            + 1,
+                                        (SysQuota::cpu_cores_quota()
+                                            * BACKGROUND_REQUEST_CORE_MAX_RATIO)
+                                            as usize,
+                                    ),
+                            );
+                        }
+
+                        if old_quota != quota_limiter.background_cputime_limiter() {
+                            info!("cpu_time_limiter tuned for backend request";
```

Comment:
Too many logs, please using debug level.

Reply:
> Too many logs, please using debug level.

Log level is temporarily switched from debug to info is only to verify the effect at testing environment. I will revert to debug level before merge.